### DISCUSSION TITLE
关于更新WIN11 24H2后出现的缩放后帧数异常问题的解决措施

### DISCUSSION THREAD

**Discussion by MOUCHEN00000001** [Blinue/Magpie#1009] [General]
### Program(Game) 游戏名

各类galgame

### Capture method 捕获模式

除了不能用的desktop capture

### Related screenshot (optional) 相关截图（可选）

![屏幕截图 2024-10-17 180529](https://github.com/user-attachments/assets/242ad1d7-2703-408c-b440-5bfabc9f5f4d)
![屏幕截图 2024-10-17 180638](https://github.com/user-attachments/assets/863f74da-3f3a-4339-8429-a3935b349e84)
![屏幕截图 2024-10-17 180649](https://github.com/user-attachments/assets/96f46e9f-98cd-4533-8210-5a9b03182848)
![屏幕截图 2024-10-18 141422](https://github.com/user-attachments/assets/320b078d-ddda-4ad3-bd39-11b858b45272)

### Other information (optional) 补充信息（可选）

具体表现为使用magpie缩放后帧数异常，然而一开magpie的内置监控，一切奇迹般地复原（画面没变化，帧数恢复至以前的正常水准，但不能在开监控的时候进行操作，何况挡视野）。我不久前重装了系统，结果发现使用magpie缩放gal时出现了帧数过低、延迟过高的问题。经过一番折腾，我终于在无意中解决（应该算是）了这个问题。出现这个问题的缘由并不是magpie的版本（我尝试了四个不同版本的magpie，出现了一样的问题），我想，罪魁祸首应该是win11的新系统。gal帧数很高（截图里跑到了六百多，榨干了我可怜的3050），gpu占用高。不知道为什么，使用magpie进行缩放后，原窗口应该还是占用了gpu的大部分算力（我尝试过切任务管理器看核显占用，但它没有任何变化，因此排除缩放后独显工作的可能）。于是，magpie无法很好地得到gpu的支持（已经被原窗口榨干力）。解决措施是锁定原窗口的帧数（游戏内或英伟达控制面板），这样就能正常使用。
我并不了解magpie的缩放原理，以上仅为个人推测，希望可以帮助遇到同样问题的人们（我已经离不开magpie了）。

### Log files 日志文件

[magpie.log](https://github.com/user-attachments/files/17430320/magpie.log)
问题解决后，复现bug失败，开监控不再能恢复帧数。我先前多次卸载magpie，直至解决问题才安装最新版本。旧的日志是从回收站扒拉出来的，应该能用。
[magpie.1.log](https://github.com/user-attachments/files/17430496/magpie.1.log)

**Comment by Blinue** [comment]
谢谢分享！Magpie 本身也是使用 GPU 执行渲染，会出现和游戏争夺资源的情况，这也是我不推荐在 3D 游戏里使用的原因之一。你的情况是游戏耗尽了 GPU 算力，这在 2D 游戏里不太常见，但有些游戏不使用垂直同步/锁帧。我不确定是游戏还是 OS 的问题，如果是更新到 24H2 后出现的，可能是 OS 或驱动的 bug。

除了在驱动里锁帧外，让游戏运行在集显上，Magpie 运行在独显上也是一个解决方案。

帧率低也和你的缩放配置有关，Anime4K GAN、Anime4K Restore 和 AdaptiveSharpen 都是很慢的。你的屏幕分辨率是 1920x1080，缩放 1280x720 的窗口不需要 Anime4K_Upscale_GAN_x3_L，它会将画面放大到三倍。

**Reply by MOUCHEN00000001** [reply]
明白了，我会尝试优化一下缩放配置。不过我的小破笔记本现在能跑40帧，而对我而言gal30帧已经绰绰有余。

**Comment by BLACKCHARGE** [comment]
这种情况很好解决，限制下程序的最大帧数就可以了，N卡控制面板里面有，一般设置成等于显示器最大刷新率就行

**Comment by BLACKCHARGE** [comment]
另外就是restore要放到upscale之前才更合适，否则会放大瑕疵，徒增功耗

**Reply by MOUCHEN00000001** [reply]
感谢，受教了

**Comment by hooke007** [comment]
![image](https://github.com/user-attachments/assets/db52652d-aa8b-4443-a292-5b92329032c2)
